src/Sulu/Bundle/SnippetBundle/Resources/js/views/SnippetAreas/SnippetAreas.js

```diff
@@ -97,9 +104,14 @@ class SnippetAreas extends React.Component<ViewProps> {
                                     <Table.Cell>
                                         {defaultUuid
                                             ? <Fragment>
-                                                <div className={snippetAreasStyles.title}>
+                                                <Button
+                                                    className={snippetAreasStyles.titleButton}
+                                                    onClick={this.handleSnippetClick}
+                                                    skin="link"
```

Comment:
I don't think that this results in the desired design, since the other deep links are also shown differently (i.e. without an underline). The other deep links are simply `div` elements with a `button` role, would keep it the same way here for consistency.

Reply:
the reason i implemented this using a `Button` is that the `react/jsx-no-bind` eslint rule forces me to extract this "simple `<div>` into a separate component. 

this is the second time in a short timespan that this rule complicates the implementation of a simple component. i would really like to enable the `allowArrowFunctions` of this it.

Reply:
Personally I don't think that this makes it a lot more complicated, and extracting separate components is most of the time not such a bad idea. Also, as [described in the rule documentation](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md#lists-of-items), that should speed up rendering. I tend to believe that, unless I see a benchmark proofing the opposite. I think we already have talked about this, and this rule would indeed not work anymore with hooks, but since we are not using them yet I would keep this rule for now.

Reply:
do you think its a good idea to extract a component in this case?
regarding performance, i tend to *dont* believe that, unless I see a benchmark proofing the opposite. Also, see https://cdb.reacttraining.com/react-inline-functions-and-performance-bdff784f5578

Reply:
TBH, although it is an unpopular opinion, I would do it, because [MobX advises to use small components](https://mobx.js.org/best/react-performance.html#use-many-small-components) (which is MobX specific, so that's for sure not mentioned in that article) and because the rule also avoids code like this:

```javascript
<div
    onClick={() =>{
        methodCall1();
        methodCall2();
        if (someValue == 5) {
            methodCall3();
        } else {
            methodCall4();
            methodCall5();
        }
    }}
/>
```

This code should IMO definitely be extracted into a separate function, and the rule also enforces this.

These two reasons alone would be enough for me to accept the minor inconvenience creating a separate component causes. But if we still decide to not do that anymore I would change that in the eslint config instead of disabling it using comments, because it has about the same effect (I do not want to think or argue at every PR if that is a valid edge case for disabling this rule).

Reply:
After some thinking this discussion took maybe a bit an unnecessary turn... I didn't dislike the `Button` solution very much, the only real concern for me was that the styling did not match. But this could be solved by introducing another skin (maybe "text"?) that would display it in the way we want it to be displayed. This way we could keep the rule, and the issue you have mentioned would be solved in many cases. What do you think?

Reply:
Yes, a `text` skin for the `Button` component sounds okay to me. It will also prevent the issues with the `react/jsx-no-bind` for the most simple cases in the future, so thats nice 🙂